Push code release failed

Run `pnpx pkg-pr-new publish`

publish error

PR:       ✔️
Push:   ❌



### logs

```
Error:  publishing failed: {"url":"/publish","statusCode":403,"statusMessage":"","message":"Resource not accessible by integration - https://docs.github.com/rest/issues/comments#create-an-issue-comment"}
  at Object.run (/home/runner/.cache/pnpm/dlx/t6c2gbxqvfjjki5ut3q3wgvbcu/193b0ccbbac-a75/node_modules/.pnpm/pkg-pr-new@0.0.35/node_modules/pkg-pr-new/dist/index.js:3421:18)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async runCommand (/home/runner/.cache/pnpm/dlx/t6c2gbxqvfjjki5ut3q3wgvbcu/193b0ccbbac-a75/node_modules/.pnpm/pkg-pr-new@0.0.35/node_modules/pkg-pr-new/dist/index.js:2196:[16](https://github.com/element-plus/element-plus/actions/runs/12257309026/job/34194482729#step:10:17))
  at async runCommand (/home/runner/.cache/pnpm/dlx/t6c2gbxqvfjjki5ut3q3wgvbcu/193b0ccbbac-a75/node_modules/.pnpm/pkg-pr-new@0.0.35/node_modules/pkg-pr-new/dist/index.js:2187:11)
  at async runMain (/home/runner/.cache/pnpm/dlx/t6c2gbxqvfjjki5ut3q3wgvbcu/193b0ccbbac-a75/node_modules/.pnpm/pkg-pr-new@0.0.35/node_modules/pkg-pr-new/dist/index.js:2323:7) 

Error:  publishing failed: {"url":"/publish","statusCode":403,"statusMessage":"","message":"Resource not accessible by integration - https://docs.github.com/rest/issues/comments#create-an-issue-comment"}
Error: Process completed with exit code 1.
```
https://github.com/element-plus/element-plus/blob/d9da5341d2ce82e2de630fc794aef0b524beb74c/.github/workflows/publish-pr-commit-pkg.yml#L51C11-L51C34
https://github.com/element-plus/element-plus/actions/runs/12257309026/job/34194482729

--- a/packages/backend/server/routes/publish.post.ts
+++ b/packages/backend/server/routes/publish.post.ts
@@ -229,7 +229,19 @@ export default eventHandler(async (event) => {
     );
 
     if (comment !== "off") {
-      if (comment === "update" && prevComment!) {
+      const { data: permissions } = await installation.request(
+        "GET /repos/{owner}/{repo}/installation",
+        {
+          owner: workflowData.owner,
+          repo: workflowData.repo,
+        }
+      );
+
+      if (permissions.permissions.issues !== 'write') {
+        console.warn(
+          `No permission to create comments in ${workflowData.owner}/${workflowData.repo}:${workflowData.ref}`,
+        );
+      } else if (comment === "update" && prevComment!) {
         await installation.request(
           "PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}",
           {

--- a/packages/backend/server/routes/publish.post.ts
+++ b/packages/backend/server/routes/publish.post.ts
@@ -229,7 +229,7 @@ export default eventHandler(async (event) => {
     );
 
     if (comment !== "off") {
-      const { data: permissions } = await installation.request(
+      const { data: { permissions } } = await installation.request(
         "GET /repos/{owner}/{repo}/installation",
         {
           owner: workflowData.owner,
@@ -237,7 +237,8 @@ export default eventHandler(async (event) => {
         }
       );
 
-      if (permissions.permissions.issues !== 'write') {
+      console.log(permissions);
+      if (permissions.issues !== 'write') {
         console.warn(
           `No permission to create comments in ${workflowData.owner}/${workflowData.repo}:${workflowData.ref}`,
         );

--- a/packages/backend/server/routes/publish.post.ts
+++ b/packages/backend/server/routes/publish.post.ts
@@ -237,51 +237,50 @@ export default eventHandler(async (event) => {
         }
       );
 
-      console.log(permissions);
-      if (permissions.issues !== 'write') {
-        console.warn(
-          `No permission to create comments in ${workflowData.owner}/${workflowData.repo}:${workflowData.ref}`,
-        );
-      } else if (comment === "update" && prevComment!) {
-        await installation.request(
-          "PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}",
-          {
-            owner: workflowData.owner,
-            repo: workflowData.repo,
-            comment_id: prevComment.id,
-            body: generatePullRequestPublishMessage(
-              origin,
-              templatesHtmlMap,
-              packagesWithoutPrefix,
-              workflowData,
-              compact,
-              onlyTemplates,
-              checkRunUrl,
-              packageManager,
-              "ref",
-            ),
-          },
-        );
-      } else {
-        await installation.request(
-          "POST /repos/{owner}/{repo}/issues/{issue_number}/comments",
-          {
-            owner: workflowData.owner,
-            repo: workflowData.repo,
-            issue_number: Number(workflowData.ref),
-            body: generatePullRequestPublishMessage(
-              origin,
-              templatesHtmlMap,
-              packagesWithoutPrefix,
-              workflowData,
-              compact,
-              onlyTemplates,
-              checkRunUrl,
-              packageManager,
-              comment === "update" ? "ref" : "sha",
-            ),
-          },
-        );
+      try {
+        if (comment === "update" && prevComment!) {
+          await installation.request(
+            "PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}",
+            {
+              owner: workflowData.owner,
+              repo: workflowData.repo,
+              comment_id: prevComment.id,
+              body: generatePullRequestPublishMessage(
+                origin,
+                templatesHtmlMap,
+                packagesWithoutPrefix,
+                workflowData,
+                compact,
+                onlyTemplates,
+                checkRunUrl,
+                packageManager,
+                "ref",
+              ),
+            },
+          );
+        } else {
+          await installation.request(
+            "POST /repos/{owner}/{repo}/issues/{issue_number}/comments",
+            {
+              owner: workflowData.owner,
+              repo: workflowData.repo,
+              issue_number: Number(workflowData.ref),
+              body: generatePullRequestPublishMessage(
+                origin,
+                templatesHtmlMap,
+                packagesWithoutPrefix,
+                workflowData,
+                compact,
+                onlyTemplates,
+                checkRunUrl,
+                packageManager,
+                comment === "update" ? "ref" : "sha",
+              ),
+            },
+          );
+        }
+      } catch (e) {
+        console.error("failed to create/update comment", e, permissions);
       }
     }
   }

